components/applications-service/integration_test/service_groups_benchmark_test.go

```diff
@@ -0,0 +1,84 @@
+//
+//  Author:: Salim Afiune <afiune@chef.io>
+//  Copyright:: Copyright 2019, Chef Software Inc.
+//
+
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/chef/automate/api/external/applications"
+	"github.com/chef/automate/api/external/habitat"
+)
+
+var (
+	sgRes *applications.ServiceGroups
+	sgErr error
+)
+
+func BenchmarkGetServiceGroups(b *testing.B) {
+	defer suite.DeleteDataFromStorage()
+
+	var (
+		ctx            = context.Background()
+		request        = new(applications.ServiceGroupsReq)
+		iterations     = 10
+		incremental    = 100
+		maxGroups      = incremental * iterations
+		nServiceGroups = 0
+	)
+
+	for i := 0; i <= iterations; i++ {
```

Comment:
AFAIK the golang benchmark suite will do as many iterations as necessary to get a meaningful result -- so what are we measuring here? Since the `b.N` loop is the innermost loop, it seems we're executing _10 runs of golang's benchmark logic_. Why are we doing that?

The alternative I'd see here is benchmarking _10 GetServiceGroups requests_, but then I'd expect the loop with `iterations` to be inside the `b.N` loop 🤔

Reply:
I am just doing multiple benchmarks with different loads. Instead of writing 10 benchmark functions, I am just doing one that tests the RPC function with loads of groups from 0 to 50,000. 

I could separate them if that is less confusing.

Reply:
Ah! [this](https://github.com/chef/automate/pull/495/files#diff-8383c9c14d839f5204cba7037be032d8R43), right? I didn't see it 😎 sorry for the noise